### ISSUE TITLE
roadmap for new docs

### ISSUE THREAD

**Issue by edublancas** [ploomber/sklearn-evaluation#107]
*Note: this is work in progress*

this document summarizes the content we'll develop for sklearn-evaluation's docs

# types of content

- quick start: get started fast
- user guide: deep dive with detailed explanations
- model diagnosis: help the user understand how to improve/fix their model
- problem-oriented: tutorials to help users improve/fix their models

# classification

## classification quick start

objective: get the user started as quickly as possible
should a short notebook with the essentials (most common plots) and simple explanations for each: confusion matrix, classification_report, roc, precision_recall

## classification user guide

objective: teach (in detail) the most critical plots for classification. at the top of the notebook should provide a summary so  users can jump into any plot (confusion matrix, classification_report, roc, precision_recall)

## classification problem-oriented notebooks

objective: should address a specific problem that a user might encounter

- model diagnosis (quickly teach the user how to understand what's wrong with the model so they can fix it. e.g., there's class imbalance, underfitting, overfitting, features with disparate ranges). these should only help diagnose. the solutions should come in a separate notebook
- class imbalance (how to assess it and how to train a model with class imbalance)
- probability estimation (describe calibration curves and scores distribution, see: https://ploomber.io/blog/calibration-curve/)

# regression

## regression quick start

objective: get the user started as quickly as possible
should have a short notebook with the essentials (most common plots) and simple explanations for each

## regression user guide

objective: teach (in detail) the most critical plots for classification. at the top of the notebook should provide a summary so  users can jump into any plot (TODO: what are the most popular plots? residuals + what others?)

## regression problem-oriented notebooks

- model diagnosis (quickly teach the user how to understand what's wrong with the model so they can fix it. e.g., outliers, heteroskedasticity, underfitting, overfitting, features with disparate ranges). these should only help diagnose. the solutions should come in a separate notebook
- finding outliers with cook's distance

# clustering

## clustering quick start

objective: get the user started as quickly as possible
should have a short notebook with the essentials (most common plots) and simple explanations for each

## clustering user guide

objective: teach (in detail) the most critical plots for classification. at the top of the notebook should provide a summary so  users can jump into any plot (TODO: what are the most popular plots? elbow?)

## clustering problem-oriented notebooks

- model diagnosis (quickly teach the user how to understand what's wrong with the model so they can fix it). these should only help diagnose. the solutions should come in a separate notebook
- visualizing clusters using PCA
- evaluating number of clusters with elbow curve

# model optimization

- tuning a hyperparameter (validation curve)
- under and over-fitting (learning curve)
- feature importance (and ranking)
- nested cross-validation (implement it with the experiment tracker)
- feature normalization (this helps with performance in a few models, but in others it has no effect, we should explain why)

**Comment by idomic** [comment]
Closing as all of those was implemented.
